server/src/main/java/org/candlepin/model/ProductAttribute.java

```diff
@@ -163,6 +163,17 @@ public int hashCode() {
     }
 
     /**
+     * Calculates and returns a version hash for this entity. This method operates much like the
+     * hashCode method, except that it is more accurate and should have fewer collisions.
+     *
+     * @return
+     *  a version hash for this entity
+     */
+    public int getEntityVersion() {
+        return this.hashCode();
```

Comment:
This is a general statement that doesn't apply to every implementation. In the cases cited here, we're retrieving the hash codes for strings, integers, booleans and entity objects which themselves define a hashCode implementation which is derived from the afore mentioned hash code implementations.

Our hash codes will be consistent from instance to instance, so long as the Java implementation itself remains consistent.